toolchain/sem_ir/inst_namer.cpp

```diff
@@ -413,70 +441,44 @@ auto InstNamer::AddBlockLabel(ScopeId scope_id, LocId loc_id, AnyBranch branch)
     AddBlockLabel(scope_id, branch.target_id, "", loc_id);
     return;
   }
-  llvm::StringRef name;
+  std::string name;
   switch (sem_ir_->parse_tree().node_kind(loc_id.node_id())) {
     case Parse::NodeKind::IfExprIf:
-      switch (branch.kind) {
-        case BranchIf::Kind:
-          name = "if.expr.then";
-          break;
-        case Branch::Kind:
-          name = "if.expr.else";
-          break;
-        case BranchWithArg::Kind:
-          name = "if.expr.result";
-          break;
-        default:
-          break;
-      }
+      name = BranchKindToName("if.expr", branch.kind, "then", "else", "result");
       break;
 
     case Parse::NodeKind::IfCondition:
-      switch (branch.kind) {
-        case BranchIf::Kind:
-          name = "if.then";
-          break;
-        case Branch::Kind:
-          name = "if.else";
-          break;
-        default:
-          break;
-      }
+      name = BranchKindToName("if", branch.kind, "then", "else",
+                              /*branch_with_arg=*/std::nullopt);
       break;
 
     case Parse::NodeKind::IfStatement:
       name = "if.done";
```

Comment:
May as well add the `<unexpected>` here too:
```suggestion
      name = BranchKindToName("if", branch.kind, {.branch = "done"});
```
(This one should never have a `BranchIf` or a `BranchWithArg` I think.)